lib/crawly/worker.ex

```diff
@@ -158,20 +160,22 @@ defmodule Crawly.Worker do
     end
   end
 
-  @spec process_parsed_item({parsed_item, response, spider_name}) :: result
+  @spec process_parsed_item({parsed_item, response, spider_name, request}) ::
+          result
         when spider_name: atom(),
+             request: Crawly.Request.t(),
              response: HTTPoison.Response.t(),
              parsed_item: Crawly.ParsedItem.t(),
              result: {:ok, :done}
-  defp process_parsed_item({parsed_item, response, spider_name}) do
+  defp process_parsed_item({parsed_item, response, spider_name, _request}) do
     requests = Map.get(parsed_item, :requests, [])
     items = Map.get(parsed_item, :items, [])
     # Process all requests one by one
     Enum.each(
       requests,
-      fn request ->
-        request = Map.put(request, :prev_response, response)
-        Crawly.RequestsStorage.store(spider_name, request)
+      fn req ->
```

Comment:
variable name change unnecessary, fully qualified names preferred for consistency.